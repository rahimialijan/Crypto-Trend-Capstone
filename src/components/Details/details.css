.details-top-details {
  display: flex;
  justify-content: space-between;
  padding: 1% 3%;
}

.details-crypto-icon img {
  width: 4rem;
}

.details-crypto-intro {
  display: flex;
  flex-direction: column;
  font-size: 1.4rem;
}

.details-crypto-analytics {
  display: flex;
  padding: 2% 3%;
  flex-wrap: wrap;
}

.details-crypto-analytics > div {
  width: 44%;
  padding: 2%;
  background-color: #5e82ae;
  display: flex;
  justify-content: space-between;
  font-size: 1.2rem;
}

.details-crypto-analytics > div:nth-child(4n),
.details-crypto-analytics > div:nth-child(4n + 1) {
  background-color: #809ec4;
}

@media only screen and (max-width: 600px) {
  .details-crypto-analytics {
    font-size: 10px;
    flex-direction: column;
  }

  .details-crypto-analytics > div {
    width: 95%;
    padding: 5% 3%;
  }

  .details-crypto-analytics > div:nth-child(even) {
    background-color: #d14378;
  }

  .details-crypto-analytics > div:nth-child(odd) {
    background-color: #dd4781;
  }
}
