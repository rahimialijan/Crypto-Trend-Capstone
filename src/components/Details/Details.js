import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useLocation } from 'react-router';
import { Link } from 'react-router-dom';
import { setTitle, setActiveCoin } from '../../redux/crypto/cryptoSlice';
import './details.css';

const Details = () => {
  const dispatch = useDispatch();
  const location = useLocation();
  // eslint-disable-next-line
  const activeData = useSelector((state) => state.crypto.cryptos.find((el) => el.uuid === location.state),);
  const date = new Date(activeData.listedAt * 1000);

  useEffect(() => {
    dispatch(setTitle(`${activeData.symbol} Treds`));
    dispatch(setActiveCoin(location.state));
  });

  return (
    <div key={activeData.uuid}>
      <div className="details-all-stats">{activeData.name.toUpperCase()}</div>
      <div className="details-top-details">
        <div className="details-crypto-icon">
          <img src={activeData.iconUrl} alt="Crypto" />
        </div>
        <div className="details-crypto-intro">
          <span>
            <span>Crypto Currency: </span>
            <span>{activeData.name}</span>
          </span>
          <span>
            <span>Price In BTC: </span>
            <span>
              {Number(activeData.btcPrice).toLocaleString(undefined, 2)}
            </span>
          </span>
          <span>
            <Link to={activeData.coinrankingUrl} target="_blank">
              External Resource
            </Link>
          </span>
        </div>
      </div>
      <div className="details-crypto-analytics">
        <div>
          <span>Name: </span>
          <span>{activeData.name}</span>
        </div>
        <div>
          <span>Volume in 24h:</span>
          <span>
            {`${Number(activeData['24hVolume']).toLocaleString(
              undefined,
              2,
            )} $`}
          </span>
        </div>
        <div>
          <span>Market Capital:</span>
          <span>
            {`${Number(activeData.marketCap).toLocaleString(undefined, 2)} $`}
          </span>
        </div>
        <div>
          <span>Price in USD: </span>
          <span>
            {`${Number(activeData.price).toLocaleString(undefined, 2)} $`}
          </span>
        </div>
        <div>
          <span>Listed At:</span>
          <span>{`${date.getFullYear()}-${date.getMonth()}-${date.getDate()}`}</span>
        </div>
        <div>
          <span>Change in 24h: </span>
          <span>{activeData.change}</span>
        </div>
        <div>
          <span>Rank: </span>
          <span>{activeData.rank}</span>
        </div>
        <div>
          <span>Low Volume: </span>
          <span>{activeData.lowVolume}</span>
        </div>
      </div>
    </div>
  );
};

export default Details;
