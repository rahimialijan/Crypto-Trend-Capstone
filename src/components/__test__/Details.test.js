import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import createMockStore from 'redux-mock-store';
import Details from '../Details/Details';

jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'),
  useLocation: () => ({
    pathname: '/details',
    search: '',
    hash: '',
    state: 'Qwsogvtv82FCd',
    key: 'default',
  }),
}));
describe('Details Test Suits', () => {
  test('Should Render Details Component', () => {
    const initialState = {
      crypto: {
        cryptos: [
          {
            uuid: 'Qwsogvtv82FCd',
            symbol: 'BTC',
            name: 'Bitcoin',
            color: '#f7931A',
            iconUrl: 'https://cdn.coinranking.com/bOabBYkcX/bitcoin_btc.svg',
            marketCap: '586699044075',
            price: '30525.981366112937',
            listedAt: 1330214400,
            tier: 1,
            change: '0.50',
            rank: 1,
            lowVolume: false,
            coinrankingUrl: 'https://coinranking.com/coin/Qwsogvtv82FCd+bitcoin-btc',
            '24hVolume': '12581186855',
            btcPrice: '1',
          },
        ],
      },
      stats: {
        total: 26788,
        totalCoins: 26788,
        totalMarkets: 36444,
        totalExchanges: 161,
        totalMarketCap: '1172476637995',
        total24hVolume: '35484231592',
      },
      title: 'Crypto Trend',
      isLoading: true,
      error: '',
      message: '',
      active: 'Qwsogvtv82FCd',
    };
    const mockStore = createMockStore();
    const mock = mockStore(initialState);
    const container = render(
      <Provider store={mock}>
        <BrowserRouter>
          <Details />
        </BrowserRouter>
        ,
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
