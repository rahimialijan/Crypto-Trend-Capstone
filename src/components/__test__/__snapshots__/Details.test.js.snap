// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Test Suits Should Render Details Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="details-all-stats"
        >
          BITCOIN
        </div>
        <div
          class="details-top-details"
        >
          <div
            class="details-crypto-icon"
          >
            <img
              alt="Crypto"
              src="https://cdn.coinranking.com/bOabBYkcX/bitcoin_btc.svg"
            />
          </div>
          <div
            class="details-crypto-intro"
          >
            <span>
              <span>
                Crypto Currency: 
              </span>
              <span>
                Bitcoin
              </span>
            </span>
            <span>
              <span>
                Price In BTC: 
              </span>
              <span>
                1
              </span>
            </span>
            <span>
              <a
                href="https://coinranking.com/coin/Qwsogvtv82FCd+bitcoin-btc"
                target="_blank"
              >
                External Resource
              </a>
            </span>
          </div>
        </div>
        <div
          class="details-crypto-analytics"
        >
          <div>
            <span>
              Name: 
            </span>
            <span>
              Bitcoin
            </span>
          </div>
          <div>
            <span>
              Volume in 24h:
            </span>
            <span>
              12,581,186,855 $
            </span>
          </div>
          <div>
            <span>
              Market Capital:
            </span>
            <span>
              586,699,044,075 $
            </span>
          </div>
          <div>
            <span>
              Price in USD: 
            </span>
            <span>
              30,525.981 $
            </span>
          </div>
          <div>
            <span>
              Listed At:
            </span>
            <span>
              2012-1-26
            </span>
          </div>
          <div>
            <span>
              Change in 24h: 
            </span>
            <span>
              0.50
            </span>
          </div>
          <div>
            <span>
              Rank: 
            </span>
            <span>
              1
            </span>
          </div>
          <div>
            <span>
              Low Volume: 
            </span>
            <span />
          </div>
        </div>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="details-all-stats"
      >
        BITCOIN
      </div>
      <div
        class="details-top-details"
      >
        <div
          class="details-crypto-icon"
        >
          <img
            alt="Crypto"
            src="https://cdn.coinranking.com/bOabBYkcX/bitcoin_btc.svg"
          />
        </div>
        <div
          class="details-crypto-intro"
        >
          <span>
            <span>
              Crypto Currency: 
            </span>
            <span>
              Bitcoin
            </span>
          </span>
          <span>
            <span>
              Price In BTC: 
            </span>
            <span>
              1
            </span>
          </span>
          <span>
            <a
              href="https://coinranking.com/coin/Qwsogvtv82FCd+bitcoin-btc"
              target="_blank"
            >
              External Resource
            </a>
          </span>
        </div>
      </div>
      <div
        class="details-crypto-analytics"
      >
        <div>
          <span>
            Name: 
          </span>
          <span>
            Bitcoin
          </span>
        </div>
        <div>
          <span>
            Volume in 24h:
          </span>
          <span>
            12,581,186,855 $
          </span>
        </div>
        <div>
          <span>
            Market Capital:
          </span>
          <span>
            586,699,044,075 $
          </span>
        </div>
        <div>
          <span>
            Price in USD: 
          </span>
          <span>
            30,525.981 $
          </span>
        </div>
        <div>
          <span>
            Listed At:
          </span>
          <span>
            2012-1-26
          </span>
        </div>
        <div>
          <span>
            Change in 24h: 
          </span>
          <span>
            0.50
          </span>
        </div>
        <div>
          <span>
            Rank: 
          </span>
          <span>
            1
          </span>
        </div>
        <div>
          <span>
            Low Volume: 
          </span>
          <span />
        </div>
      </div>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
